hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieMultiTableDeltaStreamer.java

```diff
@@ -379,23 +444,50 @@ private static String resetTarget(Config configuration, String database, String
    * Creates actual HoodieDeltaStreamer objects for every table/topic and does incremental sync.
    */
   public void sync() {
-    for (TableExecutionContext context : tableExecutionContexts) {
-      try {
-        new HoodieDeltaStreamer(context.getConfig(), jssc, Option.ofNullable(context.getProperties())).sync();
-        successTables.add(Helpers.getTableWithDatabase(context));
-      } catch (Exception e) {
-        logger.error("error while running MultiTableDeltaStreamer for table: " + context.getTableName(), e);
-        failedTables.add(Helpers.getTableWithDatabase(context));
+    if (tableExecutionContexts.isEmpty()) {
+      return;
+    }
+    if (tableExecutionContexts.get(0).getConfig().isFetchFromMultiSources) {
+      while (true) {
```

Comment:
This essentially makes it an ever running job. I guess we do not want users to have resources up and running always doing ingestion with no option of having a one time job. From what I understand you are trying to introduce a continuous mode here, which is not recommended this way. Let us make it configurable? cc @nsivabalan

Reply:
Also trying to understand this scenario. Suppose a user has a combination of target tables to be ingested, say table A and table B. Table A is bound to multiple sources, but table B depends on a single source. I hope such combinations of tables are taken care. May be we can add test cases for such combinations if not done already? @watermelon12138

Reply:
Thank you for your good advice. First, I added the parameter allowContinuousWhenMultipleSources to control whether to use the continuous mode in the case of multiple sources. Second, In the original logic, I only allowed the sink table to be unique. However, I accept your suggestion. Currently, I support multiple sink tables, and each sink table can be bound to multiple sources.